### PR TITLE
[WIP]:BZ1211727:adds sssd as auth example

### PR DESCRIPTION
**Pull request by Amrita42**
### PR creator: Description

Describe the overall goals of this pull request.
Replaces ldap wih sssd

### PR creator: Are there any relevant issues/feature requests?

[BZ1211727](https://bugzilla.suse.com/show_bug.cgi?id=1211727)

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [x] SLE 15 next/openSUSE Leap next *(current `main`, no backport necessary)*
  - [x] SLE 15 SP5/openSUSE Leap 15.5
  - [x] SLE 15 SP4/openSUSE Leap 15.4
  - [ ] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2
  - [ ] SLE 15 SP1
- SLE 12
  - [ ] SLE 12 SP5
  - [ ] SLE 12 SP4

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Comment by Amrita42** [comment]
@Firstyear could please provide a tech review.

File: xml/security_pam.xml [hunk 1]
```diff
@@ -728,7 +728,7 @@ DISPLAY     DEFAULT=${REMOTEHOST}:0.0  OVERRIDE=${DISPLAY}
      <para>
       To make sure the new authentication procedure works as planned, turn
       on debugging for all PAM-related operations. The <command>pam-config
-      --add --ldap-debug</command> turns on debugging for LDAP-related PAM
+      --add --sss-debug</command> turns on debugging for SSSD-related PAM
```
Comment (Firstyear) [inline xml/security_pam.xml:731]: This isn't how sssd debugging works. the --sss-debug option just adds debug to the pam to resolver interactions, it's not changing the debug log level in sssd itself. As a result, this section needs to be updated to reflect how to debug sssd. 

Additionally, is there any documentation here about configuring sssd.conf? 

Could someone realistically follow this chapter from start to finish and be able to log in to a system with ldap authentication? That's the barrier we have to clear.

**Review by Firstyear — CHANGES_REQUESTED** [review]

File: xml/security_pam.xml (hunk 1 above)
Comment (Amrita42) [reply xml/security_pam.xml:731]: Is this what you are referring to, as an example:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_authentication_and_authorization_in_rhel/configuring-sssd-to-use-ldap-and-require-tls-authentication_configuring-authentication-and-authorization-in-rhel
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/sssd-troubleshooting

Comment (Amrita42) [reply xml/security_pam.xml:731]: The context of this bz was to replace ldap with sssd , but yes that could be added as part of this .

File: xml/security_pam.xml [hunk 2]
```diff
@@ -751,7 +751,7 @@ DISPLAY     DEFAULT=${REMOTEHOST}:0.0  OVERRIDE=${DISPLAY}
      <para>
       Finally, remove the debug option from your setup when you are entirely
       satisfied with the performance of it. The <command>pam-config --delete
-      --ldap-debug</command> command turns off debugging for LDAP
+      --sss-debug</command> command turns off debugging for SSSD
```
Comment (Firstyear) [inline xml/security_pam.xml:754]: This is still incorrect, this is for debugging the sss pam module, not SSSD as a whole.

File: xml/security_pam.xml (hunk 1 above)
Comment (Firstyear) [reply xml/security_pam.xml:731]: Or link to a location where we actually show the full setup of how to configure sssd.

Comment (Amrita42) [reply xml/security_pam.xml:731]: I can add a section on configuring SSSD , Will tag you once I ve it ready

File: xml/security_pam.xml (hunk 2 above)
Comment (Amrita42) [reply xml/security_pam.xml:754]: Turn off/on sssd >> Add a section on configuring SSSD

Comment (Amrita42) [reply xml/security_pam.xml:754]: @Firstyear could you take a look , I think we already have a section on SSSD

Comment (Amrita42) [reply xml/security_pam.xml:754]: https://documentation.suse.com/sles/15-SP5/html/SLES-all/cha-security-ldap.html#sec-security-ldap-server-sssd

**Comment by Amrita42** [comment]
Related bz >> https://bugzilla.suse.com/show_bug.cgi?id=1205269

**Comment by Amrita42** [comment]
@Firstyear 

- Add a ref link to configure SSSD
- debugg- mentioned  the sss pam module
-  please confirm if https://bugzilla.suse.com/show_bug.cgi?id=1205269 is covered in this bz (https://bugzilla.suse.com/show_bug.cgi?id=1211727)

**Review by Firstyear — APPROVED** [review]

**Comment by Amrita42** [comment]
@dariavladykina please review . Thanks.

File: xml/security_pam.xml [hunk 3]
```diff
@@ -728,9 +728,10 @@ DISPLAY     DEFAULT=${REMOTEHOST}:0.0  OVERRIDE=${DISPLAY}
      <para>
       To make sure the new authentication procedure works as planned, turn
       on debugging for all PAM-related operations. The <command>pam-config
-      --add --ldap-debug</command> turns on debugging for LDAP-related PAM
+      --add --sss-debug</command> turns on debugging for SSSD-related PAM
```
Comment (dariavladykina) [inline xml/security_pam.xml:731]:
```suggestion
      --add --sss-debug</command> command turns on debugging for SSSD-related PAM
```

File: xml/security_pam.xml [hunk 4]
```diff
@@ -728,9 +728,10 @@ DISPLAY     DEFAULT=${REMOTEHOST}:0.0  OVERRIDE=${DISPLAY}
...
       operations.<phrase os="sles;sled;osuse"> Find the debugging output in the &systemd; journal (see
       <xref linkend="cha-journalctl"/>).</phrase>
+      <phrase os="sles;sled">To configure SSSD, refer <xref linkend="sec-security-ldap-server-sssd"/> </phrase>
```
Comment (dariavladykina) [inline xml/security_pam.xml:734]:
```suggestion
      <phrase os="sles;sled">To configure SSSD, refer to <xref linkend="sec-security-ldap-server-sssd"/>.</phrase>
```

File: xml/security_pam.xml [hunk 5]
```diff
@@ -751,8 +752,8 @@ DISPLAY     DEFAULT=${REMOTEHOST}:0.0  OVERRIDE=${DISPLAY}
      <para>
       Finally, remove the debug option from your setup when you are entirely
       satisfied with the performance of it. The <command>pam-config --delete
```
Comment (dariavladykina) [inline xml/security_pam.xml:754]:
```suggestion
      satisfied with its performance. The <command>pam-config --delete
```

**Review by dariavladykina — COMMENTED** [review]
Hi, please see some suggestions here. Thanks!

**Pull request merged**